src/autopas/containers/verletListsCellBased/verletLists/traversals/VarVerletTraversalAsBuild.h

```diff
@@ -0,0 +1,155 @@
+   * @param neighborList The neighbor list to iterate over.
+   */
+  void iterateSoA(VerletNeighborListAsBuild<Particle> &neighborList);
+
+ public:
+  /**
+   * The Constructor of VarVerletTraversalAsBuild.
+   * @param pairwiseFunctor The functor to use for the iteration.
+   */
+  explicit VarVerletTraversalAsBuild(PairwiseFunctor *pairwiseFunctor)
+      : CellPairTraversal<ParticleCell, dataLayout, useNewton3>({0, 0, 0}), _functor(pairwiseFunctor), _soa{nullptr} {}
+
+  bool getUseNewton3() const override { return useNewton3; }
+
+  DataLayoutOption getDataLayout() const override { return dataLayout; }
+
+  void initVerletTraversal(VerletNeighborListAsBuild<Particle> &neighborList) override {
+    if (dataLayout == DataLayoutOption::soa) {
+      _soa = neighborList.loadSoA(_functor);
+    }
+  }
+
+  void endVerletTraversal(VerletNeighborListAsBuild<Particle> &neighborList) override {
+    if (dataLayout == DataLayoutOption::soa) {
+      neighborList.extractSoA(_functor);
+      _soa = nullptr;
+    }
+  }
+
+  void iterateVerletLists(VerletNeighborListAsBuild<Particle> &neighborList) override {
+    switch (dataLayout) {
+      case DataLayoutOption::aos:
+        iterateAoS(neighborList);
+        break;
+      case DataLayoutOption::soa:
+        iterateSoA(neighborList);
+        break;
+      default:
+        autopas::utils::ExceptionHandler::exception("VarVerletTraversalAsBuild does not know this data layout!");
+    }
+  }
+
+  /**
+   * Empty body. Just there to fulfill CellPairTraversal interface!
+   * @param cells
+   */
+  void initTraversal(std::vector<ParticleCell> &cells) override {}
+
+  /**
+   * Empty body. Just there to fulfill CellPairTraversal interface!
+   * @param cells
+   */
+  void endTraversal(std::vector<ParticleCell> &cells) override {}
+
+  bool isApplicable() const override {
+    return dataLayout == DataLayoutOption::soa || dataLayout == DataLayoutOption::aos;
+  }
+
+  TraversalOption getTraversalType() const override { return TraversalOption::varVerletTraversalAsBuild; }
+
+ private:
+  /**
+   * The functor to use for the iteration.
+   */
+  PairwiseFunctor *_functor;
+  /**
+   * A pointer to the SoA to iterate over if DataLayout is soa.
+   */
+  SoA<typename Particle::SoAArraysType> *_soa;
+};
+
+template <class ParticleCell, class Particle, class PairwiseFunctor, DataLayoutOption dataLayout, bool useNewton3>
+void VarVerletTraversalAsBuild<ParticleCell, Particle, PairwiseFunctor, dataLayout, useNewton3>::iterateAoS(
+    VerletNeighborListAsBuild<Particle> &neighborList) {
+  const auto &list = neighborList.getInternalNeighborList();
+
+#if defined(AUTOPAS_OPENMP)
+#pragma omp parallel num_threads(list[0].size())
+#endif
+  {
+    constexpr int numColors = 8;
+    for (int c = 0; c < numColors; c++) {
+#if defined(AUTOPAS_OPENMP)
+#pragma omp for schedule(static)
+#endif
+      for (unsigned int thread = 0; thread < list[c].size(); thread++) {
+        for (const auto &pair : list[c][thread]) {
```

Comment:
Now that we have C++17 using structured bindings (https://en.cppreference.com/w/cpp/language/structured_binding) here might help readability.

Reply:
Wow, this was horribly unreadable...